src/spec.c

```diff
@@ -2888,6 +2970,15 @@ int IndexSpec_UpdateDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
     return REDISMODULE_ERR;
   }
 
+  doc.docExpirationTime.tv_nsec = doc.docExpirationTime.tv_sec = 0;
+  doc.fieldExpirations = NULL;
+  if (spec->monitorDocumentExpiration) {
+    doc.docExpirationTime = getDocExpirationTime(ctx, key);
+  }
+  if (type == DocumentType_Hash && spec->monitorFieldExpiration) {
+    doc.fieldExpirations = getHashFieldExpirationTime(spec, ctx, key);
```

Comment:
Do we need the first `getDocExpirationTime` in this case?

Reply:
yes, document expiration is not limited only to hashes.